# hCoV-19 Variant of Concern Browser
#### 410.712 Final Project
## Introduction
  
This application allows a user to explore genetic information about hCoV-19 Variants of Concern displaying various mutations with genome annotations.  A database containing the hCoV-19 genetic information, as well as distinguishing mutations, is used to provide the display data.
 
  This tool can be used in one of two ways:
  * A side-by-side comparison of two variants showing shared and unique mutations.
  * A gene-level display of all the variant mutations found on each of the genes.  
  

Due to the ever-evolving nature of this virus, this tool displays the identified variants of concern as of May 1, 2021.  The variants are denoted using their PANGO nomenclature, but countries of origin will be referenced as well (ex. "B.1.1.7(UK)").
 
## Usage:
This application requires users to be connected to the JHU bioinformatics server.  
To explore the data proceed to:

http://bfx3.aap.jhu.edu/ksteenb1/final/index.html
 
## Support:
For any bugs or operational problems, please open an issue [here](https://github.com/KSteenbergen/programming_final_project) on Github.
 
## Acknowledgements
  GISAID for access to the annotated, official reference genome of hCoV-19.  https://www.gisaid.org/  
  PANGO lineage for publishing a list of distinguishing mutations for each variant of concern. https://cov-lineages.org/
